themes/src/main/resources/theme/base/login/resources/js/menu-button-links.js

```diff
@@ -0,0 +1,328 @@
+/*
+ *   This content is licensed according to the W3C Software License at
+ *   https://www.w3.org/Consortium/Legal/2015/copyright-software-and-document
+ *
+ *   File:   menu-button-links.js
+ *
+ *   Desc:   Creates a menu button that opens a menu of links
+ */
+
+'use strict';
+
+class MenuButtonLinks {
+  constructor(domNode) {
+    this.domNode = domNode;
+    this.buttonNode = domNode.querySelector('button');
+    this.menuNode = domNode.querySelector('[role="menu"]');
+    this.menuitemNodes = [];
+    this.firstMenuitem = false;
+    this.lastMenuitem = false;
+    this.firstChars = [];
+
+    this.buttonNode.addEventListener(
+      'keydown',
+      this.onButtonKeydown.bind(this)
+    );
+    this.buttonNode.addEventListener('click', this.onButtonClick.bind(this));
+
+    var nodes = domNode.querySelectorAll('[role="menuitem"]');
```

Comment:
If you have a variable then declare it with `const` if it doesn't get re-assigned, otherwise use `let`.